### PR TITLE
[WOR-1497] Add azure control plane flag to WSM and auth with sam using an azure token if set. 

### PR DESCRIPTION
**Pull request by jsaun**
This PR adds a feature flag to flex between running on Azure and GCP, and uses a token from Azure Workload Identity instead of a Google service account to authenticate if running on Azure.

### CODE REVIEW

**Comment by sergiygetlin** [comment]
> This PR adds a feature flag to flex between running on Azure and GCP, and uses a token from Azure Workload Identity instead of a Google service account to authenticate if running on Azure.

Jonathon, do you have ticket for this?

File: service/src/main/java/bio/terra/workspace/service/iam/SamService.java [hunk 1]
```diff
@@ -149,10 +159,22 @@ private boolean isAdmin(AuthenticatedUserRequest userRequest) throws Interrupted
 
   public String getWsmServiceAccountToken() {
     try {
-      GoogleCredentials creds =
-          GoogleCredentials.getApplicationDefault().createScoped(SAM_OAUTH_SCOPES);
-      creds.refreshIfExpired();
-      return creds.getAccessToken().getTokenValue();
+      if (features.isAzureControlPlaneEnabled()) {
+        TokenCredential credential = new DefaultAzureCredentialBuilder().build();
```
Comment (sergiygetlin) [inline service/src/main/java/bio/terra/workspace/service/iam/SamService.java:163]: Maybe split implementation into 2 smaller methods here too?

File: service/src/main/java/bio/terra/workspace/app/configuration/GoogleCredentialsConfiguration.java [hunk 2]
```diff
@@ -34,4 +34,10 @@ public GoogleCredentials getGoogleCredentials() {
   public GoogleCredentials getFakeGoogleCredentials() {
     return GoogleCredentials.newBuilder().build();
   }
+
+  @Bean
+  @Profile("azure")
+  public GoogleCredentials getFakeGoogleCredentialsAzure() {
```
[the remark this answers is not in the document]
Comment (jsaun) [reply service/src/main/java/bio/terra/workspace/app/configuration/GoogleCredentialsConfiguration.java:40]: Yeah, its needed for the dependency injection. The same approach is used for unit tests, there is supposed to be a way to apply multiple profiles to the same method, something like @Profile("unittest & azure") or @Profile({"unittest", "azure"}) but I wasn't able to get it to work.

File: service/src/main/java/bio/terra/workspace/app/configuration/external/PolicyServiceConfiguration.java [hunk 3]
```diff
@@ -39,12 +51,27 @@ public String getClientCredentialFilePath() {
...
   public String getAccessToken() throws IOException {
-    try (FileInputStream fileInputStream = new FileInputStream(clientCredentialFilePath)) {
-      GoogleCredentials credentials =
-          ServiceAccountCredentials.fromStream(fileInputStream)
-              .createScoped(POLICY_SERVICE_ACCOUNT_SCOPES);
-      AccessToken token = credentials.refreshAccessToken();
-      return token.getTokenValue();
+    try {
+      if (features.isAzureControlPlaneEnabled()) {
+        TokenCredential credential = new DefaultAzureCredentialBuilder().build();
```
Comment (aherbst-broad) [inline service/src/main/java/bio/terra/workspace/app/configuration/external/PolicyServiceConfiguration.java:56]: This block seems to be repeated several times in this PR, can it be extracted to a common utility method?

File: service/src/main/java/bio/terra/workspace/app/configuration/external/BufferServiceConfiguration.java [hunk 4]
```diff
@@ -58,11 +70,26 @@ public String getClientCredentialFilePath() {
...
 
   public String getAccessToken() throws IOException {
-    try (FileInputStream fileInputStream = new FileInputStream(clientCredentialFilePath)) {
-      GoogleCredentials credentials =
-          ServiceAccountCredentials.fromStream(fileInputStream).createScoped(BUFFER_SCOPES);
-      AccessToken token = credentials.refreshAccessToken();
-      return token.getTokenValue();
+    try {
+      if (features.isAzureControlPlaneEnabled()) {
+        TokenCredential credential = new DefaultAzureCredentialBuilder().build();
```
Comment (aherbst-broad) [inline service/src/main/java/bio/terra/workspace/app/configuration/external/BufferServiceConfiguration.java:75]: Is the RBS being spun up in the azure control plane? If not, I'm wondering if we should think about throwing an exception if an attempt is made to use the RBS in azure.

File: service/src/main/java/bio/terra/workspace/common/utils/AuthUtils.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import java.io.IOException;
+import java.util.Collection;
+
+/** Helper to reduce duplicated access token retrieval code. */
+public class AuthUtils {
+  public static String GetAccessToken(boolean isAzureControlPlaneEnabled, Collection<String> scopes)
```
Comment (aherbst-broad) [inline service/src/main/java/bio/terra/workspace/common/utils/AuthUtils.java:13]: Nit: camel case:

```suggestion
  public static String getAccessToken(boolean isAzureControlPlaneEnabled, Collection<String> scopes)
```

**Review by aherbst-broad — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/app/configuration/GoogleCredentialsConfiguration.java [hunk 6]
```diff
@@ -34,4 +34,10 @@ public GoogleCredentials getGoogleCredentials() {
   public GoogleCredentials getFakeGoogleCredentials() {
     return GoogleCredentials.newBuilder().build();
   }
+
+  @Bean
+  @Profile("azure")
```
Comment (sergiygetlin) [inline service/src/main/java/bio/terra/workspace/app/configuration/GoogleCredentialsConfiguration.java:39]: @jsaun do we need to assign "unit-test" profile here?

Comment (sergiygetlin) [reply service/src/main/java/bio/terra/workspace/app/configuration/GoogleCredentialsConfiguration.java:39]: Sorry, I see you already mentioned it.

**Review by sergiygetlin — APPROVED** [review]

**Pull request merged**